intellij/src/main/java/toolWindow/MyToolWindow.java

```diff
@@ -32,20 +48,84 @@ public JComponent getversion() throws IOException {
         return versionPanel;
     }
 
+    /**
+     * this method sets the client(UI) to get input from user and generate clientclasses.
+     * @return panel
+     */
     public JComponent getInput() {
         JBPanel<JBPanel<?>> mainPanel = new JBPanel<>();
         mainPanel.setLayout(new VerticalLayout(10));
 
-        LabeledComponent<TextFieldWithBrowseButton> labeledField1 = LabeledComponent.create(new TextFieldWithBrowseButton(), "Enter the YAML file Path");
-        LabeledComponent<TextFieldWithBrowseButton> labeledField2 = LabeledComponent.create(new TextFieldWithBrowseButton(), "Enter the name of the client class");
-        LabeledComponent<TextFieldWithBrowseButton> labeledField3 = LabeledComponent.create(new TextFieldWithBrowseButton(), "Enter the name of the postclient to generate");
-        LabeledComponent<TextFieldWithBrowseButton> labeledField4 = LabeledComponent.create(new TextFieldWithBrowseButton(), "Enter the output directory");
+        // Create the TextFieldWithBrowseButton components
+        TextFieldWithBrowseButton yamlFilePathField = new TextFieldWithBrowseButton();
+        TextFieldWithBrowseButton outputpath = new TextFieldWithBrowseButton();
+        TextFieldWithBrowseButton clientClassField = new TextFieldWithBrowseButton();
+        TextFieldWithBrowseButton postClientNameField = new TextFieldWithBrowseButton();
+
+        // Input labels
+        LabeledComponent<TextFieldWithBrowseButton> yamlFilePath_label = LabeledComponent.create(yamlFilePathField, "Enter the YAML file Path");
+        LabeledComponent<TextFieldWithBrowseButton> outputpath_Label = LabeledComponent.create(outputpath, "Enter the output directory");
+        JLabel languageLabel = new JLabel("Select a Language");
+        LabeledComponent<TextFieldWithBrowseButton> clientclassname_label = LabeledComponent.create(clientClassField, "Enter a name for the client class");
+        LabeledComponent<TextFieldWithBrowseButton> clientclassnameSpace_label = LabeledComponent.create(postClientNameField, "Enter the name of the client class namespace");
+
+        JComboBox<KiotaGenerationLanguage> languageComboBox = new ComboBox<>(KiotaGenerationLanguage.values());
+        String[] include = new String[0]; //empty string
+        String[] exclude = new String[0]; //empty String
 
-        mainPanel.add(labeledField1);
-        mainPanel.add(labeledField2);
-        mainPanel.add(labeledField3);
-        mainPanel.add(labeledField4);
+        // Add an action listener to the  genrate button to perform an action when clicked
+        JButton generateButton = new JButton("Generate");
```

Comment:
is there some kind of resources/localization mechanism? this way we could put the labels there directly instead of having to do a pass later to move them

Reply:
#3109

Reply:
tracking with #3109